src/providers/node.rs

```diff
@@ -163,48 +133,89 @@ impl NodeProvider {
-        if let Some(node_pkg) = parse_regex_into_pkg(&re, node_version)? {
+        if let Some(node_pkg) = parse_regex_into_pkg(&re, node_version) {
             return Ok(Pkg::new(node_pkg.as_str()));
         }
+        println!("No node version could be matched, defaulting to {DEFAULT_NODE_PKG_NAME}");
 
         Ok(Pkg::new(DEFAULT_NODE_PKG_NAME))
     }
 
-    pub fn get_package_manager(app: &App) -> Result<String> {
+    pub fn get_package_manager(app: &App) -> String {
         let mut pkg_manager = "npm";
         if app.includes_file("pnpm-lock.yaml") {
             pkg_manager = "pnpm";
         } else if app.includes_file("yarn.lock") {
             pkg_manager = "yarn";
         }
-        Ok(pkg_manager.to_string())
+        pkg_manager.to_string()
+    }
+
+    pub fn get_install_command(app: &App) -> String {
+        let mut install_cmd = "npm i";
+        if NodeProvider::get_package_manager(app) == "pnpm" {
+            install_cmd = "pnpm i --frozen-lockfile";
+        } else if NodeProvider::get_package_manager(app) == "yarn" {
+            if app.includes_file(".yarnrc.yml") {
+                install_cmd = "yarn set version berry && yarn install --immutable --check-cache";
+            } else {
+                install_cmd = "yarn install --frozen-lockfile --production=false";
+            }
+        } else if app.includes_file("package-lock.json") {
+            install_cmd = "npm ci";
+        }
+        install_cmd.to_string()
+    }
+
+    /// Returns the nodejs nix package and the appropriate package manager nix image.
+    pub fn get_nix_packages(app: &App, env: &Environment) -> Result<Vec<Pkg>> {
+        let package_json: PackageJson = app.read_json("package.json")?;
+        let node_pkg = NodeProvider::get_nix_node_pkg(&package_json, env)?;
+
+        if NodeProvider::get_package_manager(app) == "pnpm" {
+            let mut pnpm_pkg = Pkg::new("nodePackages.pnpm");
+            // Only override the node package if not the default one
+            if node_pkg.name != *DEFAULT_NODE_PKG_NAME {
+                pnpm_pkg = pnpm_pkg.set_override("nodejs", node_pkg.name.as_str());
+            }
+            Ok(vec![node_pkg, pnpm_pkg])
+        } else if NodeProvider::get_package_manager(app) == "yarn" {
+            let mut yarn_pkg = Pkg::new("yarn");
+            // Only override the node package if not the default one
+            if node_pkg.name != *DEFAULT_NODE_PKG_NAME {
+                yarn_pkg = yarn_pkg.set_override("nodejs", node_pkg.name.as_str());
+            }
+            Ok(vec![node_pkg, yarn_pkg])
+        } else {
+            Ok(vec![node_pkg])
+        }
     }
 }
 
-fn version_number_to_pkg(version: &u32) -> Result<Option<String>> {
-    if AVAILABLE_NODE_VERSIONS.contains(version) {
-        Ok(Some(format!("nodejs-{}_x", version)))
+fn version_number_to_pkg(version: &u32) -> String {
+    if version % 2 == 1 {
+        String::from("nodejs")
```

Comment:
```suggestion
        DEFAULT_NODE_PKG_NAME.to_string()
```

Reply:
Yea good call

Reply:
`DEFAULT_NODE_PKG_NAME` is just `nodejs`?